src/thor/triplegbuilder.cc

```diff
@@ -1253,12 +1244,18 @@ void TripLegBuilder::Build(
     bool drive_on_right = graphreader.nodeinfo(start_node)->drive_on_right();
 
     // Add trip edge
-    auto trip_edge = AddTripEdge(controller, path_begin->edgeid, path_begin->trip_id, 0,
-                                 path_begin->mode, travel_types[static_cast<int>(path_begin->mode)],
-                                 mode_costing[static_cast<uint32_t>(path_begin->mode)], edge,
-                                 drive_on_right, trip_path.add_node(), tile, graphreader,
-                                 origin_second_of_week, std::abs(end_pct - start_pct), startnode.id(),
-                                 false, nullptr, path_begin->has_time_restrictions);
+    auto trip_edge =
+        AddTripEdge(controller, path_begin->edgeid, path_begin->trip_id, 0, path_begin->mode,
+                    travel_types[static_cast<int>(path_begin->mode)],
+                    mode_costing[static_cast<uint32_t>(path_begin->mode)], edge, drive_on_right,
+                    trip_path.add_node(), tile, graphreader, origin_second_of_week, startnode.id(),
+                    false, nullptr, path_begin->has_time_restrictions);
+
+    // Set length if requested. Convert to km
+    if (controller.attributes.at(kEdgeLength)) {
+      float km = std::max((edge->length() * kKmPerMeter * std::abs(end_pct - start_pct)), 0.001f);
+      trip_edge->set_length(km);
+    }
```

Comment:
now we add the length only afterwards. note that this trivial edge cannot have discontinuities/uturns so it doenst have to worry about special trimming